apps/meteor/app/api/server/v1/channels.ts

```diff
@@ -565,22 +565,36 @@ API.v1.addRoute(
 	{ authRequired: true },
 	{
 		async post() {
-			const findResult = await findChannelByIdOrName({
-				params: this.bodyParams,
-				checkedArchived: false,
-			});
+			let roomId = 'roomId' in this.bodyParams ? this.bodyParams.roomId : undefined;
+			let roomName = 'roomName' in this.bodyParams ? this.bodyParams.roomName : undefined;
 
-			const sub = await Subscriptions.findOneByRoomIdAndUserId(findResult._id, this.userId);
+			try {
+				const findResult = await findChannelByIdOrName({
+					params: this.bodyParams,
+					checkedArchived: false,
+				});
+				roomName = findResult.name;
+
+				if (findResult._id) {
+					roomId = findResult._id;
+				}
+			} catch (e) {}
```

Comment:
That is because `findChannelByIdOrName` throws an error when the room is not found

Reply:
Ohhh, got it. Still, in this case I think we'd like to throw the error as it is in this function, so we don't need a try/catch to operate on the (possibly) returned error.
Also, if we decided to use the `catch` statement to return a different error, I believe that could be considered a breaking change -- since someone could expect the error as it is thrown by `findChannelByIdOrName`

Reply:
Please use `findByTypeAndNameOrId` instead of `findChannelByIdOrName` so that you don't need  this empty try/catch, which is not accept by our lint rules.

Reply:
Lint won't allow you to have this empty catch block.
Maybe you could try using the `findOneByIdOrName` method we already have. You'd just need to add this check after it:
```
if (room.t !== 'c' && room.t !== 'l') {
	throw new Meteor.Error('error-room-not-found', 'The required "roomId" or "roomName" param provided does not match any channel');
}
```